tritium-cache/src/main/java/com/palantir/tritium/cache/Cache.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.cache;
+
+import com.google.errorprone.annotations.CheckReturnValue;
+import com.google.errorprone.annotations.CompileTimeConstant;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+public enum Cache {
+    ;
+
+    @CheckReturnValue
+    public static <K, V> NameBuilder<K, V> builder() {
+        return new CacheBuilder<>();
+    }
+
+    @CheckReturnValue
+    public interface NameBuilder<K, V> {
+
+        SizeBuilder<K, V> name(@CompileTimeConstant String name);
+    }
+
+    @CheckReturnValue
+    public interface SizeBuilder<K, V> {
+
+        ExpiryBuilder<K, V> maximumSize(long maximumSize);
+
+        ExpiryBuilder<K, V> maximumSize(long maximumSize, Weigher<? super K, ? super V> weigher);
+    }
+
+    @CheckReturnValue
+    public interface ExpiryBuilder<K, V> {
+
+        MetricsBuilder<K, V> expiry(Expiry<? super K, ? super V> expiry);
+
+        MetricsBuilder<K, V> noExpiry();
+    }
+
+    @CheckReturnValue
+    public interface MetricsBuilder<K, V> {
+
+        ExecutorBuilder<K, V> metrics(TaggedMetricRegistry taggedMetrics);
+
+        ExecutorBuilder<K, V> noMetrics();
+    }
+
+    @CheckReturnValue
+    public interface ExecutorBuilder<K, V> {
+
+        Builder<K, V> executor(ExecutorFactory executorFactory);
+    }
```

Comment:
We don't have a good way to prevent users from using undesirable executors (ex. the fork-join common pool). But at least this being a staged builder forces users to think about this. And I doubt users will reach for the fork-join common pool here unless they do intentionally want that.

Reply:
I was hoping to someday switch to virtual threads by default, but even on JDK 25 I've found it a problematic to use due to an abundance of native code (e.g. brotli compression) and being cooperatively scheduled. It is kind of odd that the current core team argues this default for `CompletableFuture` will always be ideal as they see futures as cpu-driven instead of I/O-driven work, but I think that might be Stockholm syndrome until VTs gain preemption support.